Gsoc 2019 Fourth Blog (#85)
Cantera/cantera-website: posts/GSoC_2019_Fourth_Blog.md

---
title: GSoC 2019 Third Blog
date: 2019-07-20 15:03 UTC-19:03
slug: GSoC_2019_Fourth_Blog
tags: GSoC 2019
description: 2019 Google Summer of Code Cantera Project Blog
type: text
author: Chao Xu
---

# GSoC 2019 Fourth Blog

> After two months, the functions of ChemCheck are comparatively robust. We improved editing page and ck2yaml error reporting page this week, and we

Reviewer: > comparatively robust

The features are robust compared to what? To use this phrase, there has to be a comparison. Can you try to find another word to describe what you mean here?

> After two months, the functions of ChemCheck are comparatively robust. We improved editing page and ck2yaml error reporting page this week, and we

Reviewer: ```suggestion
After two months, the functions of ChemCheck are comparatively robust. We improved the editing and ck2yaml error reporting pages this week, and we
```

> added pages for user accounts management.

Reviewer: ```suggestion
added pages for user account management.
```

<!-- TEASER_END -->

> We have included download function for the editing page, which users can download the edited file to their local directory. It will more convenient

Reviewer: ```suggestion
We have included a download function for the editing page, so users can download the edited file to their local directory. It would more convenient
```

if changes can be saved to the original file online, so that users do not have to reupload their files, but we will leave it for now since an

> editing function is not the most important part. The current exception handling for editing page is striping out all the special signs which cannot

Reviewer: > The current exception handling...

It would be good if you explained what you mean by "exception handling."

> editing function is not the most important part. The current exception handling for editing page is striping out all the special signs which cannot

Reviewer: ```suggestion
editing function is not the most important part. The current exception handling for editing page is stripping out all the characters in encodings that cannot
```

> be recognized by ace-editor, and it could also be improved by implementing some python library to guess encoding in the future. 

Reviewer: ```suggestion
be recognized by ace-editor, and it could also be improved by implementing some Python library to guess the encoding of the file in the future. 
```

Here is what the page looks like:

{{% thumbnail "/images/GSoC_2019_images/updated_editor.png" alt="Updated Editor Page" align="center" %}}

> Editor Page

Reviewer: Please reformat this so that the whole image specification appears on one line, with the caption inserted as `<p class="text-center">Editor Page</p>`. Same for the other images

{{% /thumbnail %}}

> The convert-fail page is also improved. Logging messages are added to error messages if a conversion failed. In addition, ChemCheck looks through

Reviewer: ```suggestion
The page shown when conversion fails is also improved. Logging messages are added to error messages if a conversion failed. In addition, ChemCheck looks through
```

> two reactions next to the line where the error occurs, so that users could have an idea about how to fix the error. Here is the page:

Reviewer: > two reactions

What if the problem is in a thermo entry? Is this for reactions only?

Author: It will look through three lines ahead and after the line where the error occurs.

Author: Sorry, it is actually four lines, and it is only for reactions only.

Author: I just updated the code https://github.com/comocheng/ChemCheck/pull/33, it should work for all
files now.

{{% thumbnail "/images/GSoC_2019_images/convert_fail_page.png" alt="Convert Fail Page" align="center" %}}

Convert Fail Page

{{% /thumbnail %}}

> A series of accounts management page including signup page, login page, logout page, password change page(changing password for users who has logged in), password reset page(for users who forgets password), 

Reviewer: ```suggestion
A series of account management pages including signup, login, logout, password change (changing password for users who are logged in), and password reset (for users who forget their password) pages, among others
```

> A series of accounts management page including signup page, login page, logout page, password change page(changing password for users who has logged in), password reset page(for users who forgets password), 

Reviewer: You may want to explain that signup will be optional, only necessary if a user wants to retrieve previously uploaded files. Users can still convert files without signing up for the site

> etc. have been included. A built-in django module (django.contrib.auth.urls) is implemented for this part; however, the password reset page did not work as expected, which did not send a email to user for password resetting as it supposed to.

Reviewer: ```suggestion
have been included. A built-in django module (`django.contrib.auth.urls`) is implemented for this part; however, the password reset page did not work as expected, because it did not send an email to the user to reset the password.
```

> I am working on getting this part work. Except password reset function, other pages work well.

Reviewer: ```suggestion
I am working on getting this part to work. Except the password reset function, other pages work well.
```

Signup Page:

{{% thumbnail "/images/GSoC_2019_images/signup.png" alt="Signup Page" align="center" %}}

Signup Page

{{% /thumbnail %}}

Login Page:

{{% thumbnail "/images/GSoC_2019_images/login.png" alt="Login Page" align="center" %}}

Login Page

{{% /thumbnail %}}

Password Change Page:

{{% thumbnail "/images/GSoC_2019_images/password_change.png" alt="Password Change Page" align="center" %}}

Password Change Page

{{% /thumbnail %}}

## Goals in next two weeks

As we discussed before, the website will eventually save user's uploaded files into separate folders depending on

> user id instead of primary key. Also, we will get the password-reset page work and collect various chemkin files to

Reviewer: ```suggestion
the user id instead of the primary key. Also, we will get the password-reset page work and collect various CHEMKIN files to
```

diagnose the errors and test the website.

> 

Reviewer: Couple of extra blank lines at the end here should be deleted.
